lib/facility.rb

```diff
@@ -1,14 +1,59 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles =[]
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if services.include?('Vehicle Registration')
+      vehicle.registration_date = Date.today
+      @registered_vehicles << vehicle
+
+      if vehicle.antique?
+        vehicle.plate_type = :antique
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle? == true
+        vehicle.plate_type = :ev 
+        @collected_fees += 200
+      else 
+        vehicle.plate_type = :regular
+        @collected_fees += 100
+      end
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') && registrant.permit? == true && registrant.age >= 16
+      registrant.license_data[:written] = true
+    else 
+      false 
+    end
+  end
+
+  def administer_road_test(registrant)
+    if @services.include?('Road Test') && registrant.license_data[:written] == true 
+      registrant.license_data[:license] = true 
+    else 
+      false 
+    end
+    # registrant.license_data[:license]
+  end
+
+  def renew_drivers_license(registrant)
+    if @services.include?('Renew License') && registrant.license_data[:written] == true && registrant.license_data[:license] == true
+      registrant.license_data[:renewed] = true
+    else 
+      false
+    end
+  end
 end
```

Comment:
All of these methods look great. Good job with the conditionals in each of them. You could possibly slim them up a bit by taking out conditions like `== true` because those values should either return true or false automatically. So, something like this: 

```ruby 
def administer_written_test(registrant)
    if @services.include?('Written Test') && registrant.permit? && registrant.age >= 16
      registrant.license_data[:written] = true
    else 
      false 
    end
  end

  def administer_road_test(registrant)
    if @services.include?('Road Test') && registrant.license_data[:written]
      registrant.license_data[:license] = true 
    else 
      false 
    end
    # registrant.license_data[:license]
  end

  def renew_drivers_license(registrant)
    if @services.include?('Renew License') && registrant.license_data[:written] && registrant.license_data[:license]
      registrant.license_data[:renewed] = true
    else 
      false
    end
  end
```